### PR TITLE
Test suite cleanup

### PR DESCRIPTION
**Pull request by frozencemetery**
- Ensure magtests.py reports failures
- Fix the issue with starting apache in containers
- Update the package lists to run the pkinit test
- Stop repeatedly re-opening the test log file during tests
- Various code cleanliness fixes

### CODE REVIEW

File: tests/magtests.py [hunk 1]
```diff
@@ -317,25 +311,22 @@ def setup_kdc(testdir, wrapenv):
               'KRB5_TRACE': os.path.join(testdir, 'krbtrace.log')}
     kdcenv.update(wrapenv)
 
-    with (open(testlog, 'a')) as logfile:
-        ksetup = subprocess.Popen(["kdb5_util", "create", "-W", "-s",
-                                   "-r", TESTREALM, "-P", KDC_PASSWORD],
-                                  stdout=logfile, stderr=logfile,
-                                  env=kdcenv, preexec_fn=os.setsid)
+    logfile = open(testlog, 'a')
```
Comment (simo5) [inline tests/magtests.py]: where are you closing logfile here ?

**Comment by simo5** [comment]
I am not to hot in seeing the error changes and logfile changes smushed together, what is the reason to change the logfile stuff ? Would it be possible to get that in in a separate PR? Or was it causing difficulties in getting the rest of the work done ?

File: tests/magtests.py (hunk 1 above)
Comment (frozencemetery) [reply tests/magtests.py]: We can't close `logfile` - kdcproc is using it.

**Comment by frozencemetery** [comment]
Separated out the return code changes into their own commit.  I've put it on top of the stack, and it can be moved to a separate PR if desired (but requires the error count changes, so I'm keeping it here for now).

**Comment by simo5** [comment]
So I tried this:
```
$ pylint *.py
No config file found, using default configuration
************* Module magtests
E:214, 0: Missing parentheses in call to 'print' (<string>, line 214) (syntax-error)
************* Module t_bad_acceptor_name
C:  1, 0: Missing module docstring (missing-docstring)
C: 11, 4: Invalid constant name "sess" (invalid-name)
C: 12, 4: Invalid constant name "url" (invalid-name)
C: 13, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused ST_MODE imported from stat (unused-import)
W:  7, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
C:  6, 0: standard import "from stat import ST_MODE" should be placed before "import requests" (wrong-import-order)
************* Module t_basic_k5_fail_second
C:  1, 0: Missing module docstring (missing-docstring)
C: 11, 4: Invalid constant name "s" (invalid-name)
C: 13, 4: Invalid constant name "url" (invalid-name)
C: 16, 4: Invalid constant name "r" (invalid-name)
C: 20, 4: Invalid constant name "url" (invalid-name)
C: 23, 4: Invalid constant name "r" (invalid-name)
C: 29, 4: Invalid constant name "url" (invalid-name)
C: 32, 4: Invalid constant name "r" (invalid-name)
C: 36, 4: Invalid constant name "url" (invalid-name)
C: 37, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused import sys (unused-import)
W:  7, 0: Unused HTTPBasicAuth imported from requests.auth (unused-import)
C:  6, 0: standard import "import sys" should be placed before "import requests" (wrong-import-order)
************* Module t_basic_k5
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "url" (invalid-name)
C: 11, 4: Invalid constant name "r" (invalid-name)
************* Module t_basic_k5_two_users
W: 27, 0: Bad indentation. Found 9 spaces, expected 8 (bad-indentation)
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "s" (invalid-name)
C: 12, 4: Invalid constant name "url" (invalid-name)
C: 15, 4: Invalid constant name "r" (invalid-name)
C: 19, 4: Invalid constant name "url" (invalid-name)
C: 22, 4: Invalid constant name "r2" (invalid-name)
W:  6, 0: Unused HTTPBasicAuth imported from requests.auth (unused-import)
************* Module t_basic_proxy
C: 11, 0: Wrong continued indentation (remove 2 spaces).
                                      os.environ['MAG_USER_PASSWORD'],
                                    | ^ (bad-continuation)
C: 12, 0: Wrong continued indentation (remove 2 spaces).
                                      os.environ['NSS_WRAPPER_HOSTNAME'],
                                    | ^ (bad-continuation)
C: 13, 0: Wrong continued indentation (remove 2 spaces).
                                      os.environ['WRAP_PROXY_PORT'])
                                    | ^ (bad-continuation)
C: 14, 0: No space allowed after bracket
    proxies = { "http": proxy, }
              ^ (bad-whitespace)
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "proxy" (invalid-name)
C: 14, 4: Invalid constant name "proxies" (invalid-name)
C: 15, 4: Invalid constant name "url" (invalid-name)
C: 16, 4: Invalid constant name "r" (invalid-name)
************* Module t_hostname_acceptor
C:  1, 0: Missing module docstring (missing-docstring)
C: 12, 4: Invalid constant name "sess" (invalid-name)
C: 13, 4: Invalid constant name "url" (invalid-name)
C: 14, 4: Invalid constant name "r" (invalid-name)
W:  4, 0: Unused import os (unused-import)
W:  7, 0: Unused ST_MODE imported from stat (unused-import)
W:  8, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
C:  6, 0: standard import "import sys" should be placed before "import requests" (wrong-import-order)
C:  7, 0: standard import "from stat import ST_MODE" should be placed before "import requests" (wrong-import-order)
************* Module t_nonego
C: 27, 0: Wrong continued indentation before block (add 4 spaces).
        r.headers.get("WWW-Authenticate").startswith("Negotiate")):
        ^   | (bad-continuation)
C:  1, 0: Missing module docstring (missing-docstring)
C:  9, 4: Invalid constant name "url" (invalid-name)
C: 13, 4: Invalid constant name "r" (invalid-name)
C: 23, 4: Invalid constant name "r" (invalid-name)
************* Module t_required_name_attr
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "sess" (invalid-name)
C: 11, 4: Invalid constant name "url" (invalid-name)
C: 12, 4: Invalid constant name "r" (invalid-name)
C: 17, 4: Invalid constant name "sess" (invalid-name)
C: 18, 4: Invalid constant name "url" (invalid-name)
C: 19, 4: Invalid constant name "r" (invalid-name)
C: 24, 4: Invalid constant name "sess" (invalid-name)
C: 25, 4: Invalid constant name "url" (invalid-name)
C: 26, 4: Invalid constant name "r" (invalid-name)
C: 31, 4: Invalid constant name "sess" (invalid-name)
C: 32, 4: Invalid constant name "url" (invalid-name)
C: 33, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
************* Module t_spnego_negotiate_once
C: 20, 0: Wrong continued indentation (add 4 spaces).
        r.headers.get("WWW-Authenticate").startswith("Negotiate")):
        ^   | (bad-continuation)
C: 37, 0: Trailing newlines (trailing-newlines)
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "sess" (invalid-name)
C: 11, 4: Invalid constant name "url" (invalid-name)
C: 16, 4: Invalid constant name "r" (invalid-name)
C: 25, 4: Invalid constant name "r" (invalid-name)
C: 34, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
************* Module t_spnego_no_auth
C: 21, 0: Trailing newlines (trailing-newlines)
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "sess" (invalid-name)
C: 11, 4: Invalid constant name "url" (invalid-name)
C: 13, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused HTTPKerberosAuth imported from requests_kerberos (unused-import)
W:  6, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
************* Module t_spnego_proxy
C: 28, 0: No space allowed after bracket
    proxies = { "http" : proxy, }
              ^ (bad-whitespace)
C:  1, 0: Missing module docstring (missing-docstring)
C:  9, 0: Invalid function name "getAuthToken" (invalid-name)
C:  9, 0: Missing function docstring (missing-docstring)
W:  9,17: Redefining name 'target' from outer scope (line 24) (redefined-outer-name)
C: 22, 4: Invalid constant name "s" (invalid-name)
C: 24, 4: Invalid constant name "target" (invalid-name)
C: 25, 4: Invalid constant name "url" (invalid-name)
C: 27, 4: Invalid constant name "proxy" (invalid-name)
C: 28, 4: Invalid constant name "proxies" (invalid-name)
C: 33, 4: Invalid constant name "r" (invalid-name)
C:  7, 0: standard import "from base64 import b64encode" should be placed before "import requests" (wrong-import-order)
************* Module t_spnego
E: 22, 0: invalid token (<string>, line 22) (syntax-error)
************* Module t_spnego_rewrite
C:  1, 0: Missing module docstring (missing-docstring)
C: 10, 4: Invalid constant name "sess" (invalid-name)
C: 11, 4: Invalid constant name "url" (invalid-name)
C: 12, 4: Invalid constant name "r" (invalid-name)
W:  6, 0: Unused OPTIONAL imported from requests_kerberos (unused-import)
R:  1, 0: Similar lines in 2 files
==t_basic_k5_fail_second:6
==t_basic_k5_two_users:5
from requests.auth import HTTPBasicAuth


if __name__ == '__main__':
    s = requests.Session()

    url = 'http://%s:%s@%s/basic_auth_krb5/' % (os.environ['MAG_USER_NAME'],
                                                os.environ['MAG_USER_PASSWORD'],
                                                os.environ['NSS_WRAPPER_HOSTNAME'])
    r = s.get(url)
    if r.status_code != 200: (duplicate-code)
R:  1, 0: Similar lines in 4 files
==t_required_name_attr:3
==t_spnego_negotiate_once:3
==t_spnego_no_auth:3
==t_spnego_rewrite:3
import os
import requests
from requests_kerberos import HTTPKerberosAuth, OPTIONAL


if __name__ == '__main__':
    sess = requests.Session() (duplicate-code)

-----------------------------------
Your code has been rated at 3.65/10
```

**Comment by simo5** [comment]
Given the above, maybe we split the tests fixes into a new PR and we try to squash as many pylint errors as we can and then we add pylint into travis with the proper suppressions if needed ?

**Comment by frozencemetery** [comment]
Okay.  Dropped test fixup commit from this; other PR forthcoming.

**Pull request merged**